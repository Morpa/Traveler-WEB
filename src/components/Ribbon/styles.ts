import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: absolute;
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    font-weight: ${theme.font.bold};
    color: ${theme.colors.primary};
    background-color: ${theme.colors.orange};
    font-size: ${theme.font.sizes.small};
    padding: 0 ${theme.spacings.small};
    height: 8.3rem;
    width: 5.6rem;
    top: -2rem;
    left: 3rem;
    border-radius: 1rem;

    > svg {
      width: 2.2rem;
      margin-bottom: ${theme.spacings.xxsmall};
    }

    > span {
      font-weight: ${theme.font.bold};
    }
  `}
`
