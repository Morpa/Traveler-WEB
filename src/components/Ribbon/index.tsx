import { Star } from '@styled-icons/material-outlined/Star'

import * as S from './styles'

export type RibbonProps = {
  children: React.ReactNode
  rating: number
}

const Ribbon = ({ rating, children }: RibbonProps) => (
  <S.Wrapper>
    <Star />
    <span>{rating}</span>
    {children}
  </S.Wrapper>
)

export default Ribbon
