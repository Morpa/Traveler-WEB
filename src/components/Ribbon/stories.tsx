import { Story, Meta } from '@storybook/react/types-6-0'
import Ribbon, { RibbonProps } from '.'

export default {
  title: 'Ribbon',
  component: Ribbon,
  args: {
    rating: 4.5
  }
} as Meta

export const Default: Story<RibbonProps> = (args) => (
  <div
    style={{
      marginTop: '4rem',
      width: '30rem',
      height: '25rem',
      position: 'relative',
      backgroundColor: '#888'
    }}
  >
    <Ribbon {...args} />
  </div>
)
