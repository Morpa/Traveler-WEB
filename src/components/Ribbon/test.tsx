import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Ribbon from '.'

describe('<Ribbon />', () => {
  it('should render the text correctly', () => {
    const { container } = renderWithTheme(
      <Ribbon rating={4.5}>
        <div>My city</div>
      </Ribbon>
    )

    expect(screen.getByText(/4.5/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
