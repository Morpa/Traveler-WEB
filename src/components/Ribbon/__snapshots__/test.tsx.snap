// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Ribbon /> should render the text correctly 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 600;
  color: #FFFFFF;
  background-color: #F25D27;
  font-size: 1.4rem;
  padding: 0 2.4rem;
  height: 8.3rem;
  width: 5.6rem;
  top: -2rem;
  left: 3rem;
  border-radius: 1rem;
}

.c0 > svg {
  width: 2.2rem;
  margin-bottom: 0.8rem;
}

.c0 > span {
  font-weight: 600;
}

<div
  class="c0"
>
  <svg
    aria-hidden="true"
    class="c1"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <g
      fill="none"
    >
      <path
        d="M0 0h24v24H0V0z"
      />
      <path
        d="M0 0h24v24H0V0z"
      />
    </g>
    <path
      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27z"
    />
  </svg>
  <span>
    4.5
  </span>
  <div>
    My city
  </div>
</div>
`
